ENT-270: Add enterprise message to offers page

zubair-arbi opened the pull request:
ENT-270
@asadiqbal08 @saleem-latif @mattdrayer 

Add/update the existing offers page heading message from "`Select one of the following courses.`" to "`Welcome to edX. Please choose from the courses selected by your organization to start learning.`" for **all users**.

<img width="1194" alt="screen shot 2017-04-05 at 5 30 21 pm" src="https://cloud.githubusercontent.com/assets/5072991/24705487/a05204fe-1a25-11e7-8101-92d620bee894.png">


**Previously:** 
<img width="1187" alt="screen shot 2017-03-28 at 3 25 37 pm" src="https://cloud.githubusercontent.com/assets/5072991/24602277/9005b96c-1875-11e7-8975-ff0974b4f089.png">

mattdrayer commented on ecommerce/static/templates/_offer_course_list.html:
```diff
@@ -1,7 +1,9 @@
     <% if (courses.length > 1) { %>
-    <h3 class="number-of-products-header col-xs-12"><%- gettext('Select one of the following courses.') %></h3>
+        <h3 class="offer-message-bar">
+            <%- gettext("Welcome to edX. Please choose from the courses selected by your organization to start learning.") %>
```
I'd prefer to set this string in the view and pass it via the context.  Then we can write a test to cover the string and easily add conditional logic in the future without cluttering the template code.

zubair-arbi commented:
@mattdrayer I have updated the PR to provide offer page heading from offer view and added a js test for it. Please let me know if this is what you were expecting. Also I am waiting for the Griff's suggestion on the layout/design for the message box.

mattdrayer commented:
Thanks -- I would also like to see a python test checking for the presence of the parameter in the context -- thus will help to enforce the implicit contract between view and template

asadiqbal08 commented:
👍 kindly rebase it

zubair-arbi commented:
@mattdrayer I have addressed your feedback, please review again.

mattdrayer commented:
Looks like another rebase is in order -- otherwise the PR LGTM -- 👍

@vkaracic and/or @marjev care to take a look?

marjev commented on ecommerce/coupons/views.py:
```diff
@@ -120,6 +120,11 @@ def get_context_data(self, **kwargs):
+                'offer_app_page_heading': _('Welcome to edX'),
+                'offer_app_page_heading_message': _('Please choose from the courses selected by your '
+                                                    'organization to start learning.')
+            })
```
Can you add this text in the template directly?
You can show it if there wasn't any errors in the context data.

zubair-arbi replied:
@marjev This is intentional. I have made this change as pr suggestion by @mattdrayer, this way we have covered the change in unittest and we can easily add conditional logic in the future without cluttering the template code.
